src/service/history-impl.js

```diff
@@ -602,7 +602,8 @@ export class HistoryBindingNatural_ {
     if (!state) {
       state = {};
     }
-    state[HISTORY_PROP_] = newStackIndex;
+    state = /** @type {!JsonObject} */ (
+      Object.assign({HISTORY_PROP_: newStackIndex}, state));
```

Comment:
You have this reversed, this will always receive the `HISTORY_PROP_` (also, it needs to be computed, not the literal key `"HISTORY_PROP_"`) already state. Also, I don't understand why this is used instead of simple assignment.

Reply:
Discussed offline, turns out `FakeDom` freezes history state. Restored code and skipped test with a TODO to fix.